pkg/cloudprovider/provider/gce/provider.go

```diff
@@ -265,7 +265,9 @@ func (p *Provider) Create(machine *clusterv1alpha1.Machine, data *cloudprovidert
 		Disks:             disks,
 		Labels:            labels,
 		Scheduling: &compute.Scheduling{
-			Preemptible: cfg.preemptible,
+			Preemptible:       cfg.preemptible,
+			AutomaticRestart:  &cfg.automaticRestart,
+			ProvisioningModel: cfg.provisioningModel,
```

Comment:
Check for nil before assigning these values

Reply:
what if we always provide an ok default?

Reply:
It's already going to be defaulted in the upstream API(google) if you don't provide a value. In that case I think it makes complete sense to use pointers and just rely on their defaults.